satpy/readers/avhrr_l1b_gaclac.py

```diff
@@ -163,10 +165,13 @@ def get_dataset(self, key, info):
                        'Solar contamination of blackbody in channels 3',
                        'Solar contamination of blackbody in channels 4',
                        'Solar contamination of blackbody in channels 5']
-        else:
-            data = self._get_channel(key.name)
+        elif key.name.upper() in self.chn_dict:
+            # Read and calibrate channel data
+            data = self._get_channel(name=key.name, calibration=key.calibration)
```

Comment:
You could just pass the key object to the function, maybe it'll need eg. the resolution in the future ?

Reply:
yes